File: 프로그래머스/lv2/12899. 124 나라의 숫자/124 나라의 숫자.java
import java.util.Stack;

class Solution {
    public String solution(int n) {
        String answer = "";
        Stack<Integer> answerStack = new Stack<>();
        int[] mapNumber = new int[3];
        mapNumber[0] = 4;
        mapNumber[1] = 1;
        mapNumber[2] = 2;
        
        while(n > 0){
            int rest = n % 3;
            n /= 3;
            if(rest == 0) n--;
            answerStack.push(mapNumber[rest]);
        }
        
        while(!answerStack.isEmpty()){
            answer += answerStack.pop();
        }
        
        return answer;
    }
}

File: 프로그래머스/2/12899. 124 나라의 숫자/124 나라의 숫자.py

def solution(n):
    answer = ""
    numDict = {0 : "4", 1 : "1", 2 : "2"}
    
    while(n > 0):
        rest = n % 3
        answer = numDict[rest] + answer
        if(rest == 0): n -= 1
        n //= 3    
    
    return answer